client/src/main/java/io/pravega/client/stream/ReaderGroupConfig.java

```diff
@@ -49,11 +50,45 @@
 
     private final int maxOutstandingCheckpointRequest;
 
+    private final StreamDataRetention retentionConfig;
+
+    /**
+     * If a Stream's Retention Policy {@link RetentionPolicy} is set to 'CONSUMPTION' based,
+     * the Reader Group needs to register with the Stream as a Subscriber
+     * and periodically publish retention StreamCuts either manually or at checkpoints.
+     * This can be done by setting retentionConfig to 'CONSUMPTION_BASED_USER_STREAMCUT' or
+     * 'CONSUMPTION_BASED_AT_LAST_CHECKPOINT'.
+     * If the Stream's RetentionPolicy is not Consumption based, retentionConfig should be 'NO_IMPACT'.
+     *
+     * If the Streams' Retention Policy is set to 'CONSUMPTION', but the ReaderGroup retentionConfig is set at
+     * 'NO_IMPACT', the ReaderGroup will not get registered as a Subscriber with the Stream
+     * and retention StreamCuts cannot be published for this ReaderGroup.
+     * This will break the promise of ConsumptionBased Retention.
+     *
+     * NO_IMPACT - Read Positions of Readers do not impact Stream truncation/retention.
+     * CONSUMPTION_BASED_USER_STREAMCUT - User provides the StreamCut for truncation using Consumption Based Retention.
+     * CONSUMPTION_BASED_AT_LAST_CHECKPOINT - StreamCut corresponding to lastCompletedCheckpoint is published as truncation point for Consumption Based Retention.
+     * */
+    public enum StreamDataRetention {
+        NO_IMPACT(false, false),
+        CONSUMPTION_BASED_USER_STREAMCUT(true, false),
```

Comment:
"`CONSUMPTION_BASED_USER_STREAMCUT`" reflects how it is being used. This is fine in most code, but this is our public API, so everything should reflect the user's perspective. 
So something like "HOLD_DATA_UNTIL_RELEASED" or similar depending on the name of the api they are calling (this example is assuming the method were to be called "releaseDataBefore(streamCut)" or similar).

Reply:
I do not think "HOLD_DATA_UNTIL_RELEASED" is intuitive.
Also it does not bring out the relation of this enum value to Consumption based retention.

Reply:
True. Perhaps the enum type ie `StreamDataRetention` could do this? After they all relate to that.

Reply:
Filed issue for this: https://github.com/pravega/pravega/issues/5356